@xen-orchestra/rest-api/src/users/user.controller.mts

```diff
@@ -63,4 +85,41 @@ export class UserController extends XoController<XoUser> {
   getUser(@Path() id: string): Promise<Unbrand<XoUser>> {
     return this.getObject(id as XoUser['id'])
   }
+
+  /**
+   * @example id "722d17b9-699b-49d2-8193-be1ac573d3de"
+   * @example body {
+   *   "email": "user@email.com",
+   *   "name": "updated user name",
+   *   "password": "newP4ssword",
+   *   "permission": "admin",
+   *   "preferences": {}
+   *  }
+   */
+  @Patch('{id}')
+  @Middlewares(json())
+  @SuccessResponse(noContentResp.status, noContentResp.description)
+  @Response(notFoundResp.status, notFoundResp.description)
+  @Response(featureUnauthorized.status, featureUnauthorized.description)
+  async updateUser(@Path() id: string, @Body() body: UpdateUserRequestBody): Promise<void> {
+    const { email, name, password, permission, preferences } = body
+    const isAdmin = this.restApi.xoApp.apiContext.permission === 'admin'
+    const currentUserId = this.restApi.xoApp.apiContext.user?.id
+
+    if (isAdmin) {
+      if (permission != null && id === currentUserId) {
```

Comment:
Permission cannot be `null | undefined` because you type your `UpdateUserRequestBody` with `permission?: string`
```suggestion
      if (permission !== undefined && id === currentUserId) {
```